node_modules/oae-meetups/lib/rest.js

```diff
@@ -0,0 +1,131 @@
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var log = require('oae-logger').logger('oae-meetups-rest');
+var OAE = require('oae-util/lib/oae');
+var Config = require('oae-config').config('oae-meetups');
+var MeetupsAPI = require('./api');
+var MeetupsConstants = require('oae-meetups/lib/constants').MeetupsConstants;
+
+/**
+ * @REST executeJoinApiCall
+ *
+ * Get a response for the action
+ *
+ * @Server      tenant
+ * @Method      GET
+ * @Path        /meetup/{groupId}/join
+ * @PathParam   {string}                groupId             The id of the meetup to get
+ * @HttpResponse                        301                 Redirects to BBB server meetup URL
+ * @HttpResponse                        400                 groupId must be a valid resource id
+ * @HttpResponse                        403                 The BBB feature must be enabled
+ * @HttpResponse                        401                 You are not authorized to view this meetup
+ * @HttpResponse                        404                 Could not find the specified meetup
+ */
+OAE.tenantRouter.on('get', '/api/meetup/:groupId/join', function(req, res) {
+    // Forbid action if BBB is disabled
+    if (!Config.getValue(req.ctx.tenant().alias, 'bbb', 'enabled')) {
+        res.send(403, 'Action forbiden');
+    }
+
+    MeetupsAPI.Meetups.joinMeetup(req.ctx, req.params.groupId, function(err, joinInfo) {
+        if(err) {
+            res.send(503, 'Fatal error');
+        }
+
+        if(joinInfo) {
+            //Join the meetup
+            res.writeHead(301, {Location: joinInfo.url} );
+            res.end();
+        } else {
+            res.send(401, 'You are not allowed to join this meetup');
+        }
+    });
+});
+
+/**
+ * @REST executeEndApiCall
+ *
+ * Get a response for the action
+ *
+ * @Server      tenant
+ * @Method      GET
+ * @Path        /meetup/{groupId}/end
+ * @PathParam   {string}                groupId             The id of the meetup to get
+ * @HttpResponse                        400                 groupId must be a valid resource id
+ * @HttpResponse                        401                 You are not authorized to end this meetup
+ * @HttpResponse                        403                 The BBB feature must be enabled
+ * @HttpResponse                        404                 Could not find the specified meetup
+ */
+OAE.tenantRouter.on('post', '/api/meetup/:groupId/end', function(req, res) {
+
+    // Forbid action if BBB is disabled
+    if (!Config.getValue(req.ctx.tenant().alias, 'bbb', 'enabled')) {
+        res.send(403, 'Action forbiden');
+    }
+
+    MeetupsAPI.Meetups.endMeetup(req.ctx, req.params.groupId, function(err, endInfo) {
+        if(err) {
+            // TODO Show some feedback flash message?
+            res.send(503, 'Fatal error');
+        }
+
+        // TODO Show some feedback flash message?
```

Comment:
Why don;t we release without these flash messages. They were not in the original spec but something that came up when Hassaan was implementing this code